pkg/controller/datavolume/util.go

```diff
@@ -351,6 +351,47 @@ func updateDataVolumeUseCDIPopulator(syncState *dvSyncState) {
 	cc.AddAnnotation(syncState.dvMutated, cc.AnnUsePopulator, strconv.FormatBool(syncState.usePopulator))
 }
 
+func updateDataVolumeDefaultInstancetypeLabels(client client.Client, syncState *dvSyncState) error {
+	// Skip looking anything up if any default instance type labels are already present
+	dv := syncState.dvMutated
+	for _, defaultInstanceTypeLabel := range cc.DefaultInstanceTypeLabels {
+		if _, ok := dv.Labels[defaultInstanceTypeLabel]; ok {
+			return nil
+		}
+	}
+	if dv.Spec.Source != nil && dv.Spec.Source.PVC != nil {
+		pvc := &v1.PersistentVolumeClaim{}
+		key := types.NamespacedName{
+			Name:      dv.Spec.Source.PVC.Name,
+			Namespace: dv.Spec.Source.PVC.Namespace,
+		}
+		if err := client.Get(context.TODO(), key, pvc); err != nil {
+			return err
+		}
+		copyDefaultInstancetypeLabels(dv, pvc.Labels)
+	}
+	if dv.Spec.SourceRef != nil && dv.Spec.SourceRef.Namespace != nil && dv.Spec.SourceRef.Kind == cdiv1.DataVolumeDataSource {
+		ds := &cdiv1.DataSource{}
+		key := types.NamespacedName{
+			Name:      dv.Spec.SourceRef.Name,
+			Namespace: *dv.Spec.SourceRef.Namespace,
+		}
+		if err := client.Get(context.TODO(), key, ds); err != nil {
+			return err
+		}
+		copyDefaultInstancetypeLabels(dv, ds.Labels)
+	}
```

Comment:
why check both Source and SourceRef? I guess you should return after copyDefaultInstancetypeLabels().

Reply:
We label both `DataSources` and `PVCs` so checking both is valid IMHO. I'll add the early return now.